src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -1324,3 +1311,26 @@ func (a *attachedResources) getNodeportService() *resources.NodePortServiceResou
 func (a *attachedResources) getNodeportServiceKey() types.NamespacedName {
 	return a.getNodeportService().Key()
 }
+
+func (a *attachedResources) pki() error {
+	// if already initialized, exit immediately
+	if _, ok := a.items[pki]; ok {
+		return nil
+	}
+
+	newPKI, err := certmanager.NewPki(a.ctx, a.reconciler.Client, a.cluster, a.getHeadlessServiceFQDN(), a.getClusterServiceFQDN(), a.reconciler.Scheme, a.log)
+	if err != nil {
+		return fmt.Errorf("creating pki: %w", err)
```

Comment:
if this function reports `creating pki` error context maybe it would be good to remove all other occurrences?